Example/AppDelegate+CarPlay.swift

```diff
@@ -171,6 +210,93 @@ extension AppDelegate: CarPlaySearchControllerDelegate {
+                completionHandler(strongSelf.resultsOrNoResults(results, limit: MaximumSearchResults.initial))
+            })
+        } else {
+            completionHandler(self.resultsOrNoResults(items, limit: MaximumSearchResults.initial))
+        }
+    }
+    
+    func searchTemplate(_ searchTemplate: CPSearchTemplate, selectedResult item: CPListItem, completionHandler: @escaping () -> Void) {
+        guard let items = recentSearchItems else { return }
+        let extendedItems = resultsOrNoResults(items, limit: MaximumSearchResults.extended)
+        
+        let section = CPListSection(items: extendedItems)
+        let template = CPListTemplate(title: recentSearchText, sections: [section])
+        template.delegate = self
+        pushTemplate(template, animated: true)
+    }
+    
+    func forwardGeocodeOptions(_ searchText: String) -> ForwardGeocodeOptions {
+        let options = ForwardGeocodeOptions(query: searchText)
+        options.focalLocation = CarPlaySearchController.coarseLocationManager.location
+        options.locale = Locale.autoupdatingCurrent.languageCode == "en" ? nil : .autoupdatingCurrent
+        var allScopes: PlacemarkScope = .all
+        allScopes.remove(.postalCode)
+        options.allowedScopes = allScopes
+        options.maximumResultCount = MaximumSearchResults.extended
+        options.includesRoutableLocations = true
+        return options
+    }
+    
+    func selectResult(item: CPListItem, completionHandler: @escaping () -> Void) {
+        guard let userInfo = item.userInfo as? [String: Any],
+              let placemark = userInfo[CarPlaySearchController.CarPlayGeocodedPlacemarkKey] as? NavGeocodedPlacemark,
+              let location = placemark.routableLocations?.first ?? placemark.location else {
+            completionHandler()
+            return
+        }
+        
+        recentItems.add(RecentItem(placemark))
+        recentItems.save()
+        
+        let destinationWaypoint = Waypoint(location: location, heading: nil, name: placemark.title)
+        previewRoutes(to: destinationWaypoint, completionHandler: completionHandler)
+    }
+    
+    func recentSearches(_ searchText: String) -> [CPListItem] {
+        if searchText.isEmpty {
+            return recentItems.map { $0.geocodedPlacemark.listItem() }
+        }
+        return recentItems.filter { $0.matches(searchText) }.map { $0.geocodedPlacemark.listItem() }
+    }
+    
+    @available(iOS 12.0, *)
+    func resultsOrNoResults(_ items: [CPListItem], limit: UInt?) -> [CPListItem] {
+        recentSearchItems = items
+        
+        if items.count > 0 {
+            if let limit = limit {
+                return Array<CPListItem>(items.prefix(Int(limit)))
+            }
+            
+            return items
+        } else {
+            let title = NSLocalizedString("CARPLAY_SEARCH_NO_RESULTS", bundle: .mapboxNavigation, value: "No results", comment: "Message when search returned zero results in CarPlay")
```

Comment:
This localized message moved from MapboxNavigation to Example, so the corresponding entries in the .strings files will need to be updated as well.

Alternatively, would it be possible to keep this “No results” item as a built-in fallback when `resultsOrNoResults(_:limit:)` returns `nil` or an empty array?